### PR TITLE
86 vignette update for version release v 017 of tcplfit2

### PR DESCRIPTION
**Pull request by gracezhihuizhao**
This PR covers the updates needed for the vignette listed in #86

- Example 1, added deceptions for arguments `errfunc` and `poly2.biphasic` mentioning their functionalities of specifying different error distributions and bi-phasic fitting.
- Update the plots in all examples with the new ggplot plotting functions.
- Moved the previous Example 3 which shows the use of the base R plotting functionalities to an appendix, and incorporated the introduction of plotting functionality into Example 1 and 2.
- Updated the previous Example 4 now Example 3 with more descriptive text. Demonstrate the the normalizing steps and calculations of BMAD and onesd with explicit codes instead of apply `tcpl`'s methods.

### CODE REVIEW

**Review by sedavid01 — COMMENTED** [review]
Please double check the log-scaling for the hill and gain-loss are working.

**Comment by gracezhihuizhao** [comment]
> Please double check the log-scaling for the hill and gain-loss are working.

@sedavid01 Hi Sarah, I looked into it and see where the problem is. When I generate a series of X's for plotting the line I do `conc_plot = seq(min(conc), max(conc), length = 100)` with conc in normal scale, and take `log10(conc_plot)` if log is used. However the X's are evenly spread in the normal scale does not mean they are evenly spread in the log scale. In fact like example 1 where the minimum conc is 0.03 and the maximum conc is 100, the sequence turned out to have most of the points fall between 1 and 100 and only one point falls between 0 and 1, and resulted in only one point being below 0 after taking log10, thus the linear line. So the calculation is technically not wrong but the execution and the plot can definitely be improved. I will make some updates to make sure the simulated X's are evenly taken over the range on both the normal and log10 scales.

File: vignettes/tcplfit2-vignette.Rmd
Comment (sedavid01) [inline vignettes/tcplfit2-vignette.Rmd:1]: @gracezhihuizhao, please add in how users can view the tcplfit2_core output and display what they should expect to see with some brief orientation/explanatory text. (Example 2)

**Review by sedavid01 — CHANGES_REQUESTED** [review]
Made some textual updates through Example 2.  Requesting some updates to Example 2 - i.e. addition of code chunk(s) to demonstrate how to display tcplfit2_core output as mentioned in the intro paragraph for the example.

File: vignettes/tcplfit2-vignette.Rmd
Comment (sedavid01) [inline vignettes/tcplfit2-vignette.Rmd:317]: This section of text needs to be clarified. Also, suggest splitting up the bval and pval descriptions into two sentences.

File: vignettes/tcplfit2-vignette.Rmd
Comment (sedavid01) [inline vignettes/tcplfit2-vignette.Rmd:337]: Do you really mean BMD here? Please clarify this section of text.

File: vignettes/tcplfit2-vignette.Rmd
Comment (sedavid01) [inline vignettes/tcplfit2-vignette.Rmd:344]: Please clarify/flesh-out description here. If I am a naieve reader of this vignette this may somewhat confusing.

File: vignettes/tcplfit2-vignette.Rmd
Comment (sedavid01) [inline vignettes/tcplfit2-vignette.Rmd:335]: Please double check whether BMAD is onesd is calculated as part of level-3 or level-4 in tcpl.

File: vignettes/tcplfit2-vignette.Rmd
Comment (sedavid01) [inline vignettes/tcplfit2-vignette.Rmd:288]: This needs to be removed from the vignette.  An alternative solution for displaying the updated tcpl relevant tables is necessary.

File: vignettes/tcplfit2-vignette.Rmd
Comment (sedavid01) [inline vignettes/tcplfit2-vignette.Rmd:713]: Please consider replacing hard-coded numeric values with the R code as you did else where in the vignette text. This should be feasible, but please make a note if it is not.

**Review by sedavid01 — CHANGES_REQUESTED** [review]
@gracezhihuizhao, I have reviewed the updated tcplfit2 vignette and made some textual edits.
- [x] Overall, the vignette looks good, but there were some textual edits I was unable to make as I was slightly confused what you were trying to convey.  Thus, these sections have been noted.
- [ ] Also, the tcpl connection code chunk needs to be removed from the vignette and an alternative solution needs to be put forward OR proposed.

File: vignettes/tcplfit2-vignette.Rmd
Comment (madison-feshuk) [reply vignettes/tcplfit2-vignette.Rmd:288]: Another opportunity for API integration? @cthunes @brown-jason

File: vignettes/tcplfit2-vignette.Rmd
Comment (madison-feshuk) [reply vignettes/tcplfit2-vignette.Rmd:317]: Not looking at knitted markdown, but text should be more extensive now then what's in the tcplfit2 vignette currently. Thanks to @Ashley-Ko methods clean up!

Suggest not explaining each method extensively here and instead just direct to tcpl resources...

File: vignettes/tcplfit2-vignette.Rmd
Comment (madison-feshuk) [reply vignettes/tcplfit2-vignette.Rmd:335]: Level 4 is correct

File: vignettes/tcplfit2-vignette.Rmd
Comment (madison-feshuk) [reply vignettes/tcplfit2-vignette.Rmd:337]: bmad and onesd are both set at level 4. Note these two methods will be consolidated into one in tcpl v3.2

**Review by sedavid01 — COMMENTED** [review]
I think the updates @gracezhihuizhao was able to make cover most of the textual updates requested.
@madison-feshuk and @brown-jason, I think the tcpl tables is still an open issue.  Also, may be good to have one final review of the vignette before merging to 'dev'. However, once those are complete, then this should be good to "approve" and merge to 'dev'.

**Review by madison-feshuk — APPROVED** [review]

**Comment by madison-feshuk** [comment]
@brown-jason @kpaulfriedman here is a pdf of vignettes combined. Let me know if additional edits are needed. Otherwise, package should be ready for a tech review.
[tcplfit2_vignettes_combined_26JUN2024.pdf](https://github.com/user-attachments/files/15994221/tcplfit2_vignettes_combined_26JUN2024.pdf)

**Review by brown-jason — APPROVED** [review]

**Pull request merged**
